File: piano.py
import tkinter as tk
from pygame import mixer

class VirtualPiano:
    def __init__(self, root):
        self.root = root
        self.root.title("Virtual Piano")
        self.root.geometry("800x200")

        self.notes = ["C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B", "C2"]
        self.sounds = {note: mixer.Sound(f"sounds/{note}.mp3") for note in self.notes}

        self.create_keyboard()

    def play_key(self, note):
        self.sounds[note].play()

    def create_keyboard(self):
        for note in self.notes:
            key = tk.Button(self.root, text=note, width=5, height=2, command=lambda n=note: self.play_key(n))
            key.pack(side=tk.LEFT, padx=2)

if __name__ == "__main__":
    mixer.init()
    root = tk.Tk()
    app = VirtualPiano(root)
    root.mainloop()

File: piano.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Media;
using System.Windows.Forms;

namespace VirtualPiano
{
    public partial class VirtualPianoForm : Form
    {
        private Dictionary<string, SoundPlayer> sounds = new Dictionary<string, SoundPlayer>();

        public VirtualPianoForm()
        {
            InitializeComponent();
            LoadSounds();
            CreateKeyboard();
        }

        private void LoadSounds()
        {
            string[] notes = { "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B", "C2" };

            foreach (var note in notes)
            {
                string soundPath = Path.Combine("sounds", $"{note}.wav");
                sounds[note] = new SoundPlayer(soundPath);
            }
        }

        private void CreateKeyboard()
        {
            string[] notes = { "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B", "C2" };

            foreach (var note in notes)
            {
                Button key = new Button
                {
                    Text = note,
                    Height = 100,
                    Width = note.Contains("#") ? 30 : 50,
                    BackColor = note.Contains("#") ? System.Drawing.Color.Black : System.Drawing.Color.White,
                    ForeColor = System.Drawing.Color.Black,
                    Cursor = Cursors.Hand
                };

                key.Click += (sender, e) => PlayKey(note);
                keyboardPanel.Controls.Add(key);
            }
        }

        private void PlayKey(string note)
        {
            if (sounds.TryGetValue(note, out var player))
            {
                player.Stop();
                player.Play();
            }
        }

        private void VirtualPianoForm_KeyDown(object sender, KeyEventArgs e)
        {
            string key = e.KeyData.ToString().ToUpper();
            string note = Array.Find(new[] { "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#", "A", "A#", "B", "C2" },
                                     n => n.StartsWith(key));

            if (note != null)
            {
                PlayKey(note);
                Button keyButton = keyboardPanel.Controls.Find(note, false)[0] as Button;
                keyButton.BackColor = System.Drawing.Color.Yellow;
                keyButton.Refresh();

                Timer timer = new Timer();
                timer.Interval = 300;
                timer.Tick += (s, ev) =>
                {
                    keyButton.BackColor = note.Contains("#") ? System.Drawing.Color.Black : System.Drawing.Color.White;
                    keyButton.Refresh();
                    timer.Stop();
                };

                timer.Start();
            }
        }
    }
}
